### PR TITLE
Add ubuntu 18 agent

### PR DESCRIPTION
**Pull request by timja**
I ran it with:

```json
{
  "location": "pick one",
  "resource_group_name": "****",
  "subscription_id": "***",
  "client_id": "***",
  "client_secret": "***"
}
```

```shell
packer build --var-file vars.json ubuntu-18-agent.json
```

and then tested with:

```
az vm create     --resource-group ****    --name ****     --image **** --admin-username azureuser --ssh-key-values *****
```

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      when {
+        branch 'master'
+      }
+
+      steps {
+        sh """
+          packer build \
+            --var location="East US" \
+            --var resource_group_name="packer-images" \
```
Comment (timja) [inline Jenkinsfile]:
```suggestion
            --var resource_group_name="prod-packer-images" \
```

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      when {
+        branch 'master'
+      }
+
+      steps {
+        sh """
+          packer build \
+            --var location="East US" \
+            --var resource_group_name="prod-packer-images" \
```
Comment (timja) [inline Jenkinsfile]: https://github.com/jenkins-infra/azure/pull/137

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      }
+
+      when {
+        branch 'master'
+      }
+
+      steps {
+        sh """
+          packer build \
+            --var location="East US" \
```
Comment (timja) [inline Jenkinsfile]: should this be `East US 2?`

Comment (olblak) [reply Jenkinsfile]: Yes as everything is deployed in 'east us 2'

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      steps {
+        sh 'packer validate --var-file validate-vars.json ubuntu-18-agent.json'
+      }
+    }
+
+    stage('Build') {
+      environment {
+        AZURE_SUBSCRIPTION_ID = credentials('azure-subscription-id')
+        AZURE_CLIENT_ID = credentials('azure-client-id')
+        AZURE_CLIENT_SECRET = credentials('azure-client-secret')
```
Comment (olblak) [inline Jenkinsfile]: can you add prefix 'packer' to credentials id? I'll create credentials specific for this

File: Jenkinsfile (hunk 2 above)
Comment (olblak) [reply Jenkinsfile]: that resource has been deployed

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  stages {
+    stage('Validate') {
+      steps {
+        sh 'packer validate --var-file validate-vars.json ubuntu-18-agent.json'
+      }
+    }
+
+    stage('Build') {
+      environment {
+        AZURE_SUBSCRIPTION_ID = credentials('azure-subscription-id')
```
Comment (timja) [inline Jenkinsfile]:
```suggestion
        AZURE_SUBSCRIPTION_ID = credentials('packer-azure-subscription-id')
```

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        AZURE_CLIENT_ID = credentials('azure-client-id')
```
Comment (timja) [inline Jenkinsfile]:
```suggestion
        AZURE_CLIENT_ID = credentials('packer-azure-client-id')
```

File: Jenkinsfile (hunk 4 above)
Comment (timja) [inline Jenkinsfile]:
```suggestion
        AZURE_CLIENT_SECRET = credentials('packer-azure-client-secret')
```

File: Jenkinsfile (hunk 3 above)
Comment (timja) [reply Jenkinsfile]:
```suggestion
            --var location="East US 2" \
```

**Pull request merged**